### PR TITLE
Fix `setTraceTag`

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/11615?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11615/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11615
```

</p>
</details>

### Related Issues/PRs

 #xxx

### How is this PR tested?

- [x] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests

```python
import mlflow
import time
from unittest import mock

from mlflow.entities.trace_data import TraceData

mlflow.set_tracking_uri("databricks")
name = "/Users/harutaka.kawamura@databricks.com/test1"
mlflow.set_experiment(name)
exp = mlflow.get_experiment_by_name(name)

client = mlflow.MlflowClient()
trace_info = client.create_trace(
    experiment_id=exp.experiment_id,
    timestamp_ms=123,
    execution_time_ms=567,
    status="OK",
)
client.set_trace_tag(trace_info.request_id, "key", "value")
```

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/utils/rest_utils.py [hunk 1]
```diff
@@ -215,6 +215,10 @@ def get_trace_info_endpoint(request_id):
     return f"{_REST_API_PATH_PREFIX}/mlflow/traces/{request_id}/info"
 
 
+def get_delete_tags_endpoint(request_id):
```
Comment (B-Step62) [inline mlflow/utils/rest_utils.py:218]: Should this be `get_trace_tags_endpoint`?

File: mlflow/protos/service.proto [hunk 2]
```diff
@@ -253,6 +253,19 @@ service MlflowService {
     };
   }
 
+  // Set a tag on a trace. Tags are mutable and can be updated as desired.
+  rpc setTraceTag(SetTraceTag) returns (SetTraceTag.Response) {
```
Comment (B-Step62) [inline mlflow/protos/service.proto:257]: thank you so much for the catch!!

**Review by B-Step62 — APPROVED** [review]
LGTM! Thanks for the fix!

File: mlflow/store/tracking/rest_store.py [hunk 3]
```diff
@@ -312,8 +313,8 @@ def set_trace_tag(self, request_id: str, key: str, value: str):
             key: The string key of the tag.
             value: The string value of the tag.
         """
-        req_body = message_to_json(SetTraceTag(request_id=request_id, key=key, value=value))
-        self._call_endpoint(SetTraceTag, req_body)
+        req_body = message_to_json(SetTraceTag(key=key, value=value))
+        self._call_endpoint(SetTraceTag, req_body, endpoint=get_delete_tags_endpoint(request_id))
```
Comment (harupy) [inline mlflow/store/tracking/rest_store.py:317]:
```suggestion
        self._call_endpoint(SetTraceTag, req_body, endpoint=get_set_trace_tag_endpoint(request_id))
```

File: mlflow/utils/rest_utils.py (hunk 1 above)
Comment (harupy) [inline mlflow/utils/rest_utils.py:218]:
```suggestion
def get_set_trace_tag_endpoint(request_id):
```

File: mlflow/store/tracking/rest_store.py [hunk 4]
```diff
@@ -43,6 +43,7 @@
     _REST_API_PATH_PREFIX,
     call_endpoint,
     extract_api_info_for_service,
+    get_delete_tags_endpoint,
```
Comment (harupy) [inline mlflow/store/tracking/rest_store.py:46]:
```suggestion
    get_set_trace_tag_endpoint,
```

File: mlflow/utils/rest_utils.py (hunk 1 above)
Comment (harupy) [reply mlflow/utils/rest_utils.py]: renamed!

File: tests/store/tracking/test_rest_store.py [hunk 5]
```diff
@@ -594,3 +595,26 @@ def test_delete_traces():
         )
         _verify_requests(mock_http, creds, "traces/delete-traces", "POST", message_to_json(request))
         assert res is None
+
+
+def test_delete_traces():
```
Comment (dbczumar) [inline tests/store/tracking/test_rest_store.py:600]:
```suggestion
def test_set_trace_tag():
```

**Review by dbczumar — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/11615/files#r1550782684 is addressed. Thanks @harupy !

**Pull request merged**